evaluators/rank/NdcgEvaluator/__init__.py

```diff
@@ -45,10 +45,18 @@ def evaluate(
             For detailed information, please check https://en.wikipedia.org/wiki/Discounted_cumulative_gain
         :return the evaluation metric value for the request document.
         """
+        # Information gain must be greater or equal to 0.
+        if any(item < 0 for item in actual) or any(item < 0 for item in desired):
+            raise ValueError('One or multiple score is less than 0.')
+        # Desired must in desc order.
+        if desired != sorted(desired, reverse=True):
```

Comment:
I think this is too expensive, I think we just add in docstrinh as note that it assumes that desired come ordered?

Reply:
ok